lib/server/push.api.js

```diff
@@ -568,88 +568,96 @@ Push.Configure = function(options) {
           if (isSendingNotification) {
               return;
           }
-          // Set send fence
-          isSendingNotification = true;
-
-          // var countSent = 0;
-          var batchSize = options.sendBatchSize || 1;
-
-          // Find notifications that are not being or already sent
-          var pendingNotifications = Push.notifications.find({ $and: [
-                // Message is not sent
-                { sent : { $ne: true } },
-                // And not being sent by other instances
-                { sending: { $ne: true } },
-                // And not queued for future
-                { $or: [ { delayUntil: { $exists: false } }, { delayUntil:  { $lte: new Date() } } ] }
-            ]}, {
-              // Sort by created date
-              sort: { createdAt: 1 },
-              limit: batchSize
-            });
-
-          pendingNotifications.forEach(function(notification) {
-              // Reserve notification
-              var reserved = Push.notifications.update({ $and: [
-                // Try to reserve the current notification
-                { _id: notification._id },
-                // Make sure no other instances have reserved it
-                { sending: { $ne: true } }
-              ]}, {
-                $set: {
-                  // Try to reserve
-                  sending: true
-                }
-              });
-
-              // Make sure we only handle notifications reserved by this
-              // instance
-              if (reserved) {
-
-                // Check if query is set and is type String
-                if (notification.query && notification.query === ''+notification.query) {
-                  try {
-                    // The query is in string json format - we need to parse it
-                    notification.query = JSON.parse(notification.query);
-                  } catch(err) {
-                    // Did the user tamper with this??
-                    throw new Error('Push: Error while parsing query string, Error: ' + err.message);
-                  }
-                }
-
-                // Send the notification
-                var result = Push.serverSend(notification);
-
-                if (!options.keepNotifications) {
-                    // Pr. Default we will remove notifications
-                    Push.notifications.remove({ _id: notification._id });
-                } else {
+          try {
```

Comment:
true in that case it would not be able to recover - it should probably have an exponential backoff.